### PR TITLE
Various improvements of the NVM environment

### PR DESCRIPTION
**Pull request by return42**
## What does this PR do?

Various improvements of the NVM environment, read along the commit messages to get the details.

To make things easier in the second commit a `.nvmrc` file is added and the nodejs version is pinned to v16.13.0.  By this other development tools (IDEs) are able to detect a NVM environment.

@dalf could you have a look if the new added `.nvmrc` release an issue in your HOME installation of NVM ..

The third commit shows how NVM is integrated into emacs.

In the last commit, a `./.nvm_packages` is added to install developer packages when Node.js is installed in the NVM.

### CODE REVIEW

File: .nvm_packages [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# -*- coding: utf-8; mode: conf-unix -*-
+#
+# Developement tools pre-installed in NVM's node installation [1]
+#
+# [1] https://github.com/nvm-sh/nvm#default-global-packages-from-file-while-installing
+
+eslint
```
Comment (dalf) [inline .nvm_packages:7]: I really wish to keep the node configuration in one place: 
* `https://github.com/searxng/searxng/tree/master/searx/static/themes/oscar`
* `https://github.com/searxng/searxng/tree/master/searx/static/themes/simple`

In addition what if system wide node is available?

Is there a way to keep the configuration in `.dir-locals.el`?

**Comment by dalf** [comment]
> To make things easier in the second commit a .nvmrc file is added and the nodejs version is pinned to v16.13.0. 

I'm not sure to understand why the version is pinned.

> @dalf could you have a look if the new added .nvmrc release an issue in your HOME installation of NVM ..

It works:
```
INFO:  install Node.js by NVM
Found '/home/alexandre/code/searx/.nvmrc' with version <v16.13.0>
v16.13.0 is already installed.
Now using node v16.13.0 (npm v8.1.0)
INFO:  Node.js is installed at /home/alexandre/.nvm/versions/node/v16.13.0/bin/node
INFO:  Node.js is version v16.13.0
INFO:  npm is installed at /home/alexandre/.nvm/versions/node/v16.13.0/bin/npm
INFO:  npm is version 8.1.0
INFO:  NVM is installed at /home/alexandre/.nvm
```

File: .nvm_packages (hunk 1 above)
Comment (return42) [reply .nvm_packages:7]: > I really wish to keep the node configuration in one place:

This eslint is only used in development tools like [vscode](https://marketplace.visualstudio.com/items?itemName=abumalick.vscode-nvm) or [emacs](https://github.com/rejeep/nvm.el), therefore the `.nvmrc` is needed, it does not in interfere with the eslint that is installed in the simple theme.

The idea is to have a nodejs version and development tools are installed in the project (NVM).  If you work in different projects each project can have its own environment.  It is similar to the `requirements-dev.txt`. Except that one python interpreter is installed in the OS, what is a drawback of python.

> In addition what if system wide node is available?

It does not interfere with the one from NVM / when the NVM environment is sourced, the PATH variable is modified and the NVM is in front of the PATH variable from the system.

```
$ echo $PATH
/share/searxng/.nvm/versions/node/v16.13.0/bin:/home/markus/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin ...
```

**Comment by return42** [comment]
> I'm not sure to understand why the version is pinned.

The .nvmrc is needed by the development tools .. https://github.com/searxng/searxng/pull/529#discussion_r753855057 .. 

```
INFO:  Node.js is installed at /home/alexandre/.nvm/versions/node/v16.13.0/bin/node
```

With the environment above (installed in HOME) you have your devtools defined on user-level, with the environment below, I have defined my development tools on a project level.

```
INFO:  Node.js is installed at /x/share/searxng/.nvm/versions/node/v16.13.0/bin/node
```

**Pull request merged**